packages/babel-parser/src/tokenizer/index.js

```diff
@@ -707,6 +707,22 @@ export default class Tokenizer extends LocationParser {
     }
   }
 
+  readToken_tilde(): void {
+    // '~'
+    const next = this.input.charCodeAt(this.state.pos + 1);
+    const next2 = this.input.charCodeAt(this.state.pos + 2);
```

Comment:
Can you move `next2` inside the `next is dot` branch? So we don't introduce an extra lookahead when next is not dot.